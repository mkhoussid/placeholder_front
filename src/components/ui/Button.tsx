import * as React from 'react';

const Button = React.memo((props: React.HTMLAttributes<HTMLButtonElement>) => {
	return <button {...props} />;
});

export default Button;
