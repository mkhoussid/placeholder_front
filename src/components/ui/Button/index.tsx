import * as React from 'react';
import styled from '@emotion/styled';
import Typography, { ETypographySize, ETypographyVariant } from '../Typography';
import { useStore } from 'effector-react';
import { $isMobile, $requestLoading } from 'src/features/core/effector/store';

type ButtonProps = {
	text: string;
	variant?: ETypographyVariant;
	size?: ETypographySize;
} & Omit<React.HTMLAttributes<HTMLDivElement>, 'children'>;
const Button = React.memo(
	({ text, onClick, variant = ETypographyVariant.WHITE, size = ETypographySize.LG, ...props }: ButtonProps) => {
		const isMobile = useStore($isMobile);
		const requestLoading = useStore($requestLoading);

		const handleClick = React.useCallback(
			(e: React.MouseEvent<HTMLDivElement>) => {
				e.stopPropagation();

				if (requestLoading) return;

				onClick?.(e);
			},
			[requestLoading],
		);

		return (
			<Container onClick={handleClick} isMobile={isMobile} {...props}>
				<Typography variant={variant} size={size} containerProps={{ placement: 'center' }}>
					{text}
				</Typography>
			</Container>
		);
	},
);

export default Button;

const Container = styled.div<{ isMobile: boolean }>`
	${({ isMobile }) => `
		height: auto;
		text-align: center;
		width: auto;
		cursor: pointer;
		color: #FFF;
		transition: all 0.5s;
		position: relative;
		padding: ${isMobile ? 1 : 0.5}rem;
	 
		&:before {
			content: '';
			position: absolute;
			top: 0;
			left: 0;
			width: 100%;
			height: 100%;
			z-index: 1;
			background-color: rgba(255,255,255,0.1);
			transition: all 0.3s;
		}

		&:after {
			content: '';
			position: absolute;
			top: 0;
			left: 0;
			width: 100%;
			height: 100%;
			z-index: 1;
			opacity: 0;
			transition: all 0.3s;
			border: 1px solid rgba(255,255,255,0.5);
			transform: scale(1.2,1.2);
		}

		&:hover {
			&:after {
				opacity: 1;
				transform: scale(1,1);
			}

			&:before {
				opacity: 0 ;
				transform: scale(0.5,0.5);
			}
		}
	`}
`;
