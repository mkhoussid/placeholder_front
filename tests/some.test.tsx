import * as React from 'react';
import renderer from 'react-test-renderer';
import Button from '../src/components/ui/Button';

function toJson(component: renderer.ReactTestRenderer) {
	const result = component.toJSON();
	expect(result).toBeDefined();
	expect(result).not.toBeInstanceOf(Array);
	return result as renderer.ReactTestRendererJSON;
}

test('Link changes the class when hovered', () => {
	const component = renderer.create(<Button>Hello</Button>);
	const tree = toJson(component);
	expect(tree).toMatchSnapshot();
});
