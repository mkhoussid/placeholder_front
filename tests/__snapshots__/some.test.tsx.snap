// Vitest Snapshot v1

exports[`Link changes the class when hovered 1`] = `
<button>
  Hello
</button>
`;
